src/uu/hashsum/src/hashsum.rs

```diff
@@ -297,7 +297,7 @@ pub fn uumain(mut args: impl uucore::Args) -> UResult<()> {
     };
     let check = matches.contains_id("check");
     let tag = matches.contains_id("tag");
-    let nonames = if binary_name == "b3sum" {
+    let nonames = if binary_name == "b3sum" || matches.try_contains_id("b3sum").is_ok() {
```

Comment:
I think checking whether it is `Ok` is not enough. `try_contains_id` returns `Ok` when the `id` is valid (i.e. there is an `Arg` in `Command` with that `id`). We also need it to be present, so `Ok(true)`, otherwise `nonames` can be true when an algorithm other than `b3sum` is used (I put a `dbg!(nonames)` on line 305):

```
❯ cargo run -- hashsum --b2sum --no-names
[src/uu/hashsum/src/hashsum.rs:305] nonames = true
```

Note that we can assume that `b3sum` is a valid id, so `matches.contains_id("b3sum)` should be good enough.

Reply:
I used `try_contains_id` because it panics in test_hashsum::test_check_sha1 due to 
```
thread 'test_hashsum::test_check_sha1' panicked at 'Command was expected to succeed.
stdout =
 stderr = thread 'main' panicked at 'Mismatch between definition and access of `b3sum`. Unknown argument or group id.  Make sure you are using the argument id and not the short or long flags
```

I thought about using `try_contains_id("b3sum").unwrap_or(false)`, but when I run `cargo run -- hashsum --b2sum --no-names` `nonames` is false and it's unexpected.

If `nonames` should be set for all other algorithms according to the `no-names` argument, how about this?

```
let nonames = matches.try_contains_id("no-names").
```

Reply:
Oh I see! try_contains is correct then but we need `matches.try_contains("b3sum") == Some(true)`.

Reply:
Okay! Thanks! I'll work on it 😃👍🏻